pennylane/transforms/broadcast_expand.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module contains the tape expansion function for expanding a
+broadcasted tape into multiple tapes."""
+import pennylane as qml
+from .batch_transform import batch_transform
+
+
+@batch_transform
+def broadcast_expand(tape):
+    r"""Expand a broadcasted tape into multiple tapes
+    and a function that stacks and squeezes the results.
+
+    Args:
+        tape (.QuantumTape): Broadcasted tape to be expanded
+
+    Returns:
+        tuple[list[.QuantumTape], function]: Returns a tuple containing a list of
+        quantum tapes that produce one of the results of the broadcasted tape each,
+        and a function that stacks and squeezes the tape execution results.
+
+    This expansion function is used internally whenever a device does not
+    support broadcasting.
+
+    **Example**
+
+    We may use ``broadcast_expand`` manually on a tape to separate it
+    into multiple calculations. For this we will provide ``qml.RX`` with
+    the ``ndim_params`` attribute that allows the operation to detect
+    broadcasting.
+
+    >>> qml.RX.ndim_params = (0,)
+    >>> with qml.tape.QuantumTape() as tape:
+    >>>     qml.RX(np.array([0.2, 0.6, 1.0], requires_grad=True), wires=0)
+    >>>     qml.expval(qml.PauliZ(0))
+    >>> tapes, fn = qml.transforms.broadcast_expand(tape)
+    >>> tapes
+    [<QuantumTape: wires=[0], params=1>,
+     <QuantumTape: wires=[0], params=1>,
+     <QuantumTape: wires=[0], params=1>]
+    >>> fn(qml.execute(tapes, qml.device("default.qubit", wires=1), None))
+    array([0.98006658, 0.82533561, 0.54030231])
```

Comment:
Only one minor suggestion @dwierichs --- could you show a QNode example first, before the tape example?

With the QNode example, you can also use `qml.draw()` to show the QNode producing multiple circuits :)

Reply:
The reason for this is just to 'overcommunicate' that this works with QNodes, otherwise we have developers dropping down to tapes when they don't need to